cartesi-rollups/build-dapps/create-dapp.md

```diff
@@ -5,13 +5,345 @@ tags: [build, dapps ,developer]
+
+3. Next, we needed to add the `FORTUNE_CMD` command to our code as follows:
+
+```python
+FORTUNE_CMD = "/usr/games/fortune; exit 0"
+```
+
+This command sets the FORTUNE_CMD variable to a string that, when executed, runs the fortune program to display a random quote and then immediately exits the shell with a success status.
+
+
+4. We then replaced the existing `try` function of `def handle_advance(data)` with the following command that calls the `fortune` app:
+
+```python
+    try:
+        input = hex2str(data["payload"])
+        logger.info(f"Received input: {input}")
+
+        quote = subprocess.check_output(FORTUNE_CMD, shell=True, stderr=subprocess.STDOUT).decode()
+        output = f"Received input: {input}. This was the quote {quote}"
+
+        # Emits notice with result of calculation
+        logger.info(f"Adding notice with payload: '{output}'")
+        response = requests.post(rollup_server + "/notice", json={"payload": str2hex(str(output))})
+        logger.info(f"Received notice status {response.status_code} body {response.content}")
+```
+
+The `quote` runs the command stored in FORTUNE_CMD using a shell, captures the output (including errors), and then decodes it from bytes to a string, storing the result in the variable output. The `output` creates a formatted string with placeholders, replacing {input} and {quote} with the values of the variables input and quote, respectively, and assigns the resulting string to the variable output.
+
+
+5. We then changed the code of the `def handle_inspect(data):` function to:
+
+```python
+    logger.info(f"Received inspect request data {data}")
+
+    status = "accept"
+    try:
+        input = hex2str(data["payload"])
+        logger.info(f"Received input: {input}")
+
+        output = subprocess.check_output(FORTUNE_CMD, shell=True, stderr=subprocess.STDOUT).decode()
+
+        # Emits notice with result of calculation
+        logger.info(f"Adding report with payload: '{output}'")
+        response = requests.post(rollup_server + "/report", json={"payload": str2hex(str(output))})
+        logger.info(f"Received report status {response.status_code} body {response.content}")
+
+    except Exception as e:
+        status = "reject"
+        msg = f"Error processing data {data}\n{traceback.format_exc()}"
+        logger.error(msg)
+        response = requests.post(rollup_server + "/report", json={"payload": str2hex(msg)})
+        logger.info(f"Received report status {response.status_code} body {response.content}")
+
+    return status
+```
+
+### Build the application
+
+From `my-app` directory:
+
+```shell
+sunodo build
+```
+
+### Start the application
+
+From `my-app` directory:
+
+```shell
+sunodo run
+```
+
+### Interact with the application
+
+Now that our application is running, we can make an inspect call using the `Inspect` endpoint running at `http://localhost:8080/inspect/`.
+
+As a result we will receive payload with a witty quote, similar to the below:
```

Comment:
If we just go to http://localhost:8080/inspect/ in the browser we get a Hex and not the de-codified payload. To make an inspect call we have to use the front end console here. I Think there should be a link jumping to the section from here.